tritium-processor/src/main/java/com/palantir/tritium/processor/TritiumAnnotationProcessorStrategy.java

```diff
@@ -0,0 +1,282 @@
+import com.palantir.delegate.processors.AnnotatedTypeMethod;
+import com.palantir.delegate.processors.DelegateProcessorStrategy;
+import com.palantir.delegate.processors.LocalVariable;
+import com.palantir.tritium.annotations.Instrument;
+import com.palantir.tritium.annotations.internal.InstrumentationBuilder;
+import com.palantir.tritium.api.event.InstrumentationFilter;
+import com.palantir.tritium.event.Handlers;
+import com.palantir.tritium.event.InvocationContext;
+import com.palantir.tritium.event.InvocationEventHandler;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeMirror;
+import javax.tools.Diagnostic.Kind;
+
+enum TritiumAnnotationProcessorStrategy implements DelegateProcessorStrategy {
+    INSTANCE;
+
+    private static final String FILTER_NAME = "filter";
+    private static final String HANDLER_NAME = "handler";
+    private static final String CONTEXT_NAME = "_invocationContext";
+
+    @Override
+    public Set<String> supportedAnnotations() {
+        return Set.of(Instrument.class.getName());
+    }
+
+    @Override
+    public String generatedTypeName(String annotatedTypeName) {
+        return "Instrumented" + annotatedTypeName;
+    }
+
+    @Override
+    public TypeName delegateType(DelegateTypeArguments arguments) {
+        TypeElement type = arguments.type().type();
+        TypeName typeName = TypeName.get(type.asType());
+
+        if (type.getKind() != ElementKind.INTERFACE) {
+            arguments
+                    .context()
+                    .messager()
+                    .printMessage(
+                            Kind.ERROR,
+                            "Only interfaces may be instrumented using @" + Instrument.class.getSimpleName(),
+                            arguments.type().type());
+            return TypeName.get(type.asType());
+        }
+
+        List<? extends TypeMirror> interfaces = type.getInterfaces();
+        if (interfaces.size() != 1) {
+            return typeName;
+        }
+
+        if (type.getEnclosedElements().stream().anyMatch(element -> !isOverride(element))) {
+            return typeName;
+        }
```

Comment:
This check exists to support cases where you want to partially instrument an external interface. See the added `DelegateToRunnableMethod` and `DelegateToCallableMethod` tests for an example.

This feels a little loose, but I think will be fine in practice. The Java compiler disallows adding the `@Override` annotation to a method that does not override a superclass method.

This will not work if your delegate interface contains additional elements (ex. static fields), but the current implementation doesn't support that either, which feels reasonable.